civiform/civiform
# Fix TI seeing client's name in top nav login state

### Description

This fixes a bug wherein the Trusted Intermediary incorrectly sees their client's (applicant's) name in the upper right corner for "Logged in as _____". There is already a banner at the top stating "You are applying for [first], [last]".

This bugfix displays the TI's email address instead. Why not their name? We have no guarantee that the TI's name is available, most of the time it won't be. Their email address should always be available though.

**Technical note: consistent email on CiviFormProfile**

A tricky aspect of fixing this was making the TI's email address available in `ApplicantLayout`. The available state in the layout class includes the TI's profile and the applicant's `applicant` record. Adding new params to the already param-heavy render method would have involved updating many call sites. One option could have been to load the TI's `account` record using their profile, but this would have necessitated making a blocking DB call in the middle of the view code.

The alternative taken here involves updating CiviForm's profile creation logic to ensure all `CiviFormProfileData` (the data serialized into the pac4j session) have the email address for their account stored under the `email` attribute. This was previously inconsistent since different identity providers store the email under different attributes, e.g. IDCS stores it under `user_emailid`.

By ensuring all authenticated profiles have an email under `email` we can use the `getEmail()` method on the parent class, `CommonProfile`.

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Test with IDCS on Seattle staging
- [ ] Test with auth0 on aws-staging

### Instructions for manual testing

1. Login as a Trusted Intermediary using an identity provider (NOT using the dev tools link)
1. Navigate to a client's "Applicant Dashboard"
1. Confirm that the Trusted Intermediary's email address appears in the upper right corner instead of the client's (applicant's) name

### Issue(s) this completes

Fixes https://github.com/civiform/civiform/issues/5054
